### DISCUSSION TITLE
Why does Next.js make it such a PITA to disable SSR?

### DISCUSSION THREAD

**Discussion by punkpeye** [vercel/next.js#40699] [Help]
I love Next.js but I do not want to SSR my pages.

Why is there not a straightforward configuration to disable all SSR?

Instead I have to wrap my pages in dynamic imports. It feels ugly and it also comes with a ton of edge cases where it breaks.

**Comment by icyJoseph** [comment]
I guess, to globally disable it, the most naive way is to implement this in `_app`:

```tsx
import "../styles/globals.css";
import type { AppProps } from "next/app";
import { useState, useEffect } from "react";

function MyApp({ Component, pageProps }: AppProps) {
  const [csr, setCSR] = useState(false);

  useEffect(() => {
    setCSR(typeof window !== "undefined");
  }, []);

  return csr ? <Component {...pageProps} /> : null;
}

export default MyApp;
```

Then the body element in the upfront html (view-source) looks like:

```html
<body><div id="__next"></div><script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{}},"page":"/","query":{},"buildId":"ZZTIDe_hkm3FVDIQqY_Wp","nextExport":true,"autoExport":true,"isFallback":false,"scriptLoader":[]}</script></body>
```

Though I am not sure if that's what you want, nor if there's edge cases on this situation.
